services/dydx_client/withdrawal.py

```diff
@@ -27,11 +31,37 @@ def slow_withdrawal(self, withdrawal_params):
         return vars(withdrawal)
 
     def fast_withdrawal(self, withdrawal_params):
-        fast_withdrawal = self.client.private.create_fast_withdrawal(
-            **withdrawal_params
+        withdrawal_amount = withdrawal_params["withdrawal_amount"]
+        fast_withdrawal_result = self.get_transfer(withdrawal_amount)
+        lp_position_id_result = list(
+            fast_withdrawal_result.data["liquidityProviders"].keys()
+        )[0]
+        quote = fast_withdrawal_result.data["liquidityProviders"][
+            lp_position_id_result
+        ]["quote"]
+        if quote is None:
+            raise Exception("Could not get a quote")
+        debit_amount = quote["debitAmount"]
+        create_fast_withdrawal_result = self.client.private.create_fast_withdrawal(
+            position_id=withdrawal_params["position_id"],
+            credit_asset=constants.ASSET_USDC,
+            credit_amount=withdrawal_amount,
+            debit_amount=debit_amount,
+            to_address=withdrawal_params["to_address"],
+            lp_position_id=lp_position_id_result,
+            lp_stark_public_key=list(
+                fast_withdrawal_result.data["liquidityProviders"].values()
+            )[0]["starkKey"],
+            expiration=epoch_seconds_to_iso(time.time() + 604801),
         )
-        return vars(fast_withdrawal)
+        return create_fast_withdrawal_result.data
 
     def transfer(self, params):
         transfers = self.client.private.get_transfers(**params)
         return vars(transfers)
+
+    def get_transfer(self, withdrawal_amount):
+        get_fast_withdrawal_result = self.client.public.get_fast_withdrawal(
+            creditAsset=constants.ASSET_USDC, creditAmount=withdrawal_amount
+        )
+        return get_fast_withdrawal_result
```

Comment:
what is this function doing? Getting us the status of the fast_withdrawal right ? Whether its PENDING or CONFIRMED

Reply:
if yes, then do you really think `get_transfer` is the right name for the function ?

Reply:
no actually  it will return us the status that do dydx have quote  to make this transaction  means does dydx have liquidity  providers  to fill the our fast withdrawal

Reply:
what ??

Reply:
can you post an example response ?

Reply:
`
{'data': {'liquidityProviders': {'2': {'availableFunds': '928754.560621', 'starkKey': '04a9ecd28a67407c3cff8937f329ca24fd631b1d9ca2b9f2df47c7ebf72bf0', 'quote': {'creditAsset': 'USDC', 'creditAmount': '100', 'debitAmount': '100.470000'}}}}`

Reply:
okay and what are we doing with this result other than just returning it ? as in are we checking availableFunds value and checking if **withdrawal_amount** is <= **availableFunds** or something ?

Reply:
`  fast_withdrawal_result = self.get_transfer(withdrawal_amount)
        lp_position_id_result = list(
            fast_withdrawal_result.data["liquidityProviders"].keys()
        )[0]
        quote = fast_withdrawal_result.data["liquidityProviders"][
            lp_position_id_result
        ]["quote"]
        if quote is None:
            raise Exception("Could not get a quote")
        debit_amount = quote["debitAmount"]
` we are doing this much stuff from this getting debit amount , lp postion id , and also the

Reply:
> okay and what are we doing with this result other than just returning it ? as in are we checking availableFunds value and checking if **withdrawal_amount** is <= **availableFunds** or something ?

we don't. need to checkout for the available  fund if we don't have enough  funds  it will handless  by the apis it self with this im getting the debit amount + lp postion id

Reply:
this task we are doing after getting the transfer info [https://github.com/CruizeFinance/trident/pull/4/files#diff-dea75d7a344f892f8d402ef6aadc41257519d13fb5210474a99adfb5fbe05215R35](url)

Reply:
[https://github.com/CruizeFinance/trident/pull/4/files#diff-dea75d7a344f892f8d402ef6aadc41257519d13fb5210474a99adfb5fbe05215R35](url)